### PR TITLE
Added service wrapper docs

### PR DESCRIPTION
**Pull request by ThomasJRyan**

### CODE REVIEW

File: docs/connections/wrapper.rst [hunk 1]
```diff
@@ -0,0 +1,308 @@
+.. _service_wrapper::
+  
+Service Wrapper
+===============
+
+The Service Wrapper is a class designed to wrap various methods of a Connection object with pre, post, and exception handlers, adding extra functionality to existing Connection methods. It allows users to extend and customize the behavior of specific Connection methods without modifying the original implementation.
```
Comment (dwapstra) [inline docs/connections/wrapper.rst:6]: Please wrap to max line length of 80 characters.

File: docs/connections/wrapper.rst [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+
+4) (Optional) Implement the necessary methods in the child class to define pre, post, and exception handling behavior. These methods will be executed at different stages during the service call.
+
+5) (Optional) Use Testcase `Steps` to provide more robust reporting and leverage the pyATS reporter functionality.
+
+6) (Optional) Configure CLI parser arguments in the child class using the `configure_parser` method. This is used to enable easy integration with CLI interfaces.
+
+.. note:: 
+
+    error handlers can suppress an exception, and/or track/register it
```
Comment (dwapstra) [inline docs/connections/wrapper.rst:39]:
```suggestion
    Error handlers can suppress an exception, and/or track/register it 
```

File: docs/connections/wrapper.rst [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+        with steps.start('Call Service Step') as step:
+            step.passed('Sucessfully ran call service step')
+
+    def exception_service(self, e, steps, *args, **kwargs) -> None:
+        with steps.start('Exception Service Step') as step:
+            step.passed('Sucessfully ran exception service step')
+
+.. note:: The `steps` argument is only filled when the service wrapper is run in the context of a Testcase. If no Testcase is found, the `steps` argument will be `None`. Keep this in mind if your service wrapper is used in a standalone context with no Testcase.
+
+Execution
```
Comment (dwapstra) [inline docs/connections/wrapper.rst:273]: Discovery

File: docs/connections/hook.rst
Comment (dwapstra) [inline docs/connections/hook.rst:1]: Update changelog to explain hooks have been replaced by service wrappers.

**Review by Lelor — APPROVED** [review]

**Review by lsheikal — APPROVED** [review]

**Pull request merged**